provider/cloudflare/cloudflare.go

```diff
@@ -825,6 +856,23 @@ func (p *CloudFlareProvider) newCloudFlareChange(action string, ep *endpoint.End
 			RegionKey: regionKey,
 		}
 	}
+
+	comment := p.DNSRecordsConfig.Comment
+	if val, ok := ep.GetProviderSpecificProperty(annotations.CloudflareRecordCommentKey); ok {
+		comment = val
+	}
+
+	if len(comment) > freeZoneMaxCommentLength {
+		free := !p.ZoneHasPaidPlan(ep.DNSName)
+		if free {
+			log.Warnf("DNS record comment is invalid. Trimming comment of %s. To avoid endless syncs, please set it to less than %d chars.", ep.DNSName, freeZoneMaxCommentLength)
+			comment = comment[:freeZoneMaxCommentLength-1]
+		} else if len(comment) > paidZoneMaxCommentLength {
+			log.Warnf("DNS record comment is invalid. Trimming comment of %s. To avoid endless syncs, please set it to less than %d chars.", ep.DNSName, paidZoneMaxCommentLength)
+			comment = comment[:paidZoneMaxCommentLength-1]
+		}
+	}
```

Comment:
Any chance to move this logic under struct? To reduce complexity of current method?

Example

```go
func (cfg DNSRecordsConfig) trimAndValidateComment( isPaidZone bool, comment, dnsName string) string {
	if len(comment) > freeZoneMaxCommentLength {
		if !isPaidZone {
			log.Warnf("DNS record comment is invalid. Trimming comment of %s. To avoid endless syncs, please set it to less than %d chars.", dnsName, freeZoneMaxCommentLength)
			return comment[:freeZoneMaxCommentLength-1]
		} else if len(comment) > paidZoneMaxCommentLength {
			log.Warnf("DNS record comment is invalid. Trimming comment of %s. To avoid endless syncs, please set it to less than %d chars.", dnsName, paidZoneMaxCommentLength)
			return comment[:paidZoneMaxCommentLength-1]
		}
	}
	return comment
}
```

Reply:
updated! :) also improved the new test cases by iterating comment cases and covering all possibilities.